pkg/controller/job/job_controller.go

```diff
@@ -783,6 +804,11 @@ func (jm *Controller) manageJob(job *batch.Job, activePods []*v1.Pod, succeeded
 		rmAtLeast = 0
 	}
 	podsToDelete := activePodsForRemoval(job, activePods, int(rmAtLeast))
+	podsDeleteNumPerSync := len(podsToDelete)
+	if podsDeleteNumPerSync > MaxPodCreateDeletePerSync {
```

Comment:
this can be simplified a bit:

```golang
if len(podsToDelete) > MaxPodCreateDeletePerSync {
  podsToDelete = podsToDelete[:MaxPodCreateDeletePerSync]
}
```